docs/developer-docs/integrations/sns/managing-sns/sns-asset-canister.md

```diff
@@ -0,0 +1,146 @@
+
+The contents of the asset canister must be configured prior to the launch of the SNS, and any changes afterwards must be made by a principal with the permission to do so. The SNS is the only one that can update the list of approved principals that have this permission, and any additions to this list approved on via an SNS proposal. This configuration assures that changes to the asset canister are only made by authorized principals that the SNS has agreed on through proposals. These changes are referred to as 'upgrades' to the asset canister in the remainder of this document. 
+
+When a dapp's control is handed over to an SNS, this is also true for the associated asset canister.
+
+This section is relevant if your project contains an asset canister and describes how you can test handing over control of an asset canister to an SNS.
+
+## Deploying an asset canister
+
+To deploy an asset canister, first you will need an `assets.wasm.gz` file and a `candid/assets.did` file. Examples of these files can be downloaded with the commands:
+
+```
+curl -L "https://github.com/dfinity/sdk/raw/${DX_COMMIT}/src/distributed/assetstorage.wasm.gz" -o assets.wasm.gz
+curl -L "https://raw.githubusercontent.com/dfinity/sdk/${DX_COMMIT}/src/distributed/assetstorage.did" -o candid/assets.did
+```
+
+Next, you will need to configure your `dfx.json` file to specify your asset canister. An example `dfx.json` file can be found below:
+
+```
+{
+  "canisters": {
+    "nns-dapp": {
+      "type": "custom",
+      "candid": "",
+      "wasm": "nns-dapp.wasm"
+    },
+    "internet_identity": {
+      "type": "custom",
+      "candid": "",
+      "wasm": "internet_identity.wasm"
+    },
+    "sns_aggregator": {
+      "type": "custom",
+      "candid": "candid/sns_aggregator.did",
+      "wasm": "sns_aggregator.wasm"
+    },
+    "test": {
+      "candid": "candid/test.did",
+      "type": "rust",
+      "package": "test"
+    },
+    "assets": {
+      "build": "",
+      "candid": "candid/assets.did",
+      "type": "custom",
+      "wasm": "./assets.wasm.gz"
+    }
+  }
+}
+```
+
+### Deploying locally
+
+To deploy your asset canister locally for testing purposes, the following command can be used:
+
+```
+dfx deploy assets --network "local" --no-wallet
+```
+
+Once the asset canister has been deployed, the SNS can be launched. You can learn more about launching an SNS [here](../launch-sns/launch-sns.md).
+
+### Deploying on the mainnet
+
+:::info 
+To deploy to the mainnet network, you will need a wallet that contains cycles. For more information on cycles wallets, please see [here](../../../setup/cycles/cycles-wallet.md).
+:::
+
+To deploy your asset canister to the mainnet, the following command can be used:
+
+```
+dfx deploy assets --network "ic" --wallet <principal>
+```
+
+Once the asset canister has been deployed, the SNS can be launched. You can learn more about launching an SNS [here](../launch-sns/launch-sns.md)
+
```

Comment:
1. In the end of the deployment, would it make sense to link to docs that explain how the assets can be upgraded as long as the asset cansiter is deployed but not yet handed over to an SNS?

2. Between the "deployment" and the "submitting an SNS proposal"..  I would  maybe have a section saying what a dev has to do with the asset canister during the SNS launch (hand it over, set the roles correctly so that afterwards everything works, register the correct generic proposals).